sql/branch.sql

```diff
@@ -0,0 +1,29 @@
+-- Alter the enums to cater for missing data.
+ALTER TYPE payment_net ADD VALUE 'unknown';
+ALTER TYPE exchange_status ADD VALUE 'unknown';
+
+-- Update the field status in the exchanges table from NULL to 'unknown'
+UPDATE exchanges SET status = 'unknown' WHERE status IS NULL;
+
+-- Alter the exchanges table to ensure that no more NULL values are entered
+ALTER TABLE exchanges ALTER COLUMN status SET NOT NULL;
+
+-- Insert records for ‘unknown’ (previously NULL in exchanges table
+-- network in exchange_route table
+INSERT INTO  exchange_routes (participant, network, address, error)
+     (
+       SELECT DISTINCT participants.id, 'unknown'::payment_net, 'None', 'None'
```

Comment:
There's a uniqueness constraint on `(participant, network, address)`, which is why non-unique `(network, address)` is fine.

```sql
gratipay-bak=# \d exchange_routes
                                Table "public.exchange_routes"
┌─────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│   Column    │     Type      │                          Modifiers                           │
├─────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ id          │ integer       │ not null default nextval('exchange_routes_id_seq'::regclass) │
│ participant │ bigint        │ not null                                                     │
│ network     │ payment_net   │ not null                                                     │
│ address     │ text          │ not null                                                     │
│ error       │ text          │ not null                                                     │
│ fee_cap     │ numeric(35,2) │                                                              │
└─────────────┴───────────────┴──────────────────────────────────────────────────────────────┘
Indexes:
    "exchange_routes_pkey" PRIMARY KEY, btree (id)
    "exchange_routes_participant_network_address_key" UNIQUE CONSTRAINT, btree (participant, network, address)
Check constraints:
    "exchange_routes_address_check" CHECK (address <> ''::text)
Foreign-key constraints:
    "exchange_routes_participant_fkey" FOREIGN KEY (participant) REFERENCES participants(id)
Referenced by:
    TABLE "exchanges" CONSTRAINT "exchanges_route_fkey" FOREIGN KEY (route) REFERENCES exchange_routes(id)
```